rust/lance-linalg/src/distance/norm_l2.rs

```diff
@@ -30,21 +30,40 @@ pub trait Normalize<T: Float> {
     fn norm_l2(&self) -> Self::Output;
 }
 
+// `avx512fp16` is not supported in rustc yet. Once it is supported, we can
+// move it to target_feture.
+#[cfg(any(
+    all(target_os = "macos", target_feature = "neon"),
+    feature = "avx512fp16"
+))]
+mod kernel {
+    use super::*;
+
+    extern "C" {
+        pub fn norm_l2_f16(ptr: *const f16, len: u64) -> f16;
```

Comment:
You use `size_t` in the C code but `u64` here. Should make those consistent.

Reply:
ok, modified it to u32